hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/HFileOutputFormat2.java

```diff
@@ -286,12 +286,9 @@ public void write(ImmutableBytesWritable row, V cell)
           configureStoragePolicy(conf, fs, tableAndFamily, writerPath);
         }
 
-        if (wl != null && wl.written + length >= maxsize) {
-          this.rollRequested = true;
-        }
-
         // This can only happen once a row is finished though
-        if (rollRequested && Bytes.compareTo(this.previousRow, rowKey) != 0) {
+        if (wl != null && wl.written + length >= maxsize
```

Comment:
Make a method that checks null and written length since you have this text twice now?

Reply:
> Make a method that checks null and written length since you have this text twice now?

Sorry，this only appear in this line once，I haven't found the second line？

Reply:
You are right. My mistake.